exporter/terra/spreadsheet/handler.py

```diff
@@ -30,7 +30,7 @@ def handle_message(self, body: dict, msg: Message):
         message = SpreadsheetExporterMessage(body)
         self.logger.info('Received spreadsheet export message, informing ingest')
         self.ingest.set_spreadsheet_generation(message.job_id, ExportContextState.STARTED)
-        self.exporter.export_spreadsheet(message.job_id, message.project_uuid, message.submission_uuid)
+        self.exporter.export_spreadsheet(message.project_uuid, message.submission_uuid)
```

Comment:
What is the reason for removing job_id?
I thought it is important for the logs.

Reply:
The [spreadsheet handler](https://github.com/ebi-ait/ingest-exporter/blob/b2891181687c240bd9b57628ab91b3115dd91312/exporter/terra/spreadsheet/handler.py#L19) has a `set_context` method which it overrides from the Abstract Base Class [MessageHandler](https://github.com/ebi-ait/ingest-exporter/blob/b2891181687c240bd9b57628ab91b3115dd91312/exporter/queue/handler.py#L8) used by all our handler classes.

```handler.py
def set_context(self, body: dict) -> SessionContext:
    return SessionContext(
        logger=self.logger,
        context={
            'submission_uuid': body.get('submissionUuid'),
            'export_job_id': body.get('exportJobId'),
            'project_uuid': body.get('projectUuid')
        }
    )
```
Our generic queue Listener calls the `set_context` method on the instance of whatever [MessageHandler](https://github.com/ebi-ait/ingest-exporter/blob/b2891181687c240bd9b57628ab91b3115dd91312/exporter/queue/handler.py#L8) is provided in the [`try_handle_or_reject` method](https://github.com/ebi-ait/ingest-exporter/blob/b2891181687c240bd9b57628ab91b3115dd91312/exporter/queue/listener.py#L34).
```
def try_handle_or_reject(self, body: str, msg: Message):
    json_body: dict = json.loads(body)
    with self.handler.set_context(json_body) as s:
        s.logger.info(f'Message received')
        try:
            self.handler.handle_message(json_body, msg)
        except Exception as e:
            s.logger.error(f"Rejecting message: {body} due to error: {str(e) if str(e) else e.__class__.__name__}")
            msg.reject(requeue=False)
            s.logger.exception(e)
```
This ensures:

1. *All logging* to the specified logger contains the context.
2. Whenever an uncaught exception is thrown the message is rejected and logged with the very same context.
3. Which means that generically:
    - The concern of the listener to listen for a message and to set the context for that message and guarding against exceptions thrown while processing the message
    - The concern of the handler is coordination with the rest of the system (job started, job finished)
    - The concern of the exporter is actually doing the export

Reply:
With that preamble. All logs from the spreadsheet handler, spreadsheet exporter, will have the right context as long as the get the correct logger by name.

I can even extend that support to the terra and GCS clients if we pass them the correct logger name. I'll shortly make a commit to the terra storage client to demonstrate this.

Reply:
If the job id is logged due to the context, we are good.